packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/ItemCharts/StockDistribution.tsx

```diff
@@ -112,18 +113,56 @@ const MonthlyBar = ({
   );
 };
 
+const CalculationError = ({
+  isAmcZero,
+  isSohAndQtyZero,
+}: {
+  isAmcZero?: boolean;
+  isSohAndQtyZero?: boolean;
+}) => {
+  const t = useTranslation('replenishment');
+  const detail = isAmcZero
+    ? `: ${t('error.amc-is-zero')}`
+    : isSohAndQtyZero
+    ? `: ${t('error.soh-and-suggested-quantity-are-zero')}`
+    : '';
+  const message = `${t('error.unable-to-calculate')}${detail}`;
+
+  return (
+    <Box
+      sx={{
+        paddingLeft: 4,
+        paddingRight: 4,
+        paddingTop: 4,
+        paddingBottom: 2,
+      }}
+    >
+      <Typography variant="body1" fontWeight={700} style={{ marginBottom: 10 }}>
+        {t('heading.stock-distribution')}
+      </Typography>
```

Comment:
Since this container and heading is duplicated for both the Chart and the Error -- how hard would it be to put the conditional inside the `Box` rather than doubling up all the styling and layout?

Ignore me if that's too much hassle to re-factor. 😉

Reply:
Thanks! yeah, good idea. have done.
It also prompted me to have a look at why the OK & Next was so slow.. so that's better now too 😁